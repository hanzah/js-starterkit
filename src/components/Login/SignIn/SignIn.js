import React from 'react';
import Login from '../LoginLayout/Desktop';
import Logo from '../shared/LoginLogo';
import Form from './SignInForm';

const SignIn = () => {
  return (
    <div data-spec='sign-in-screen'>
      <Login logo={<Logo />} form={<Form />} />
    </div>
  );
}

export default SignIn;
