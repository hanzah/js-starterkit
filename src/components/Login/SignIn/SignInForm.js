import React, { Component, PropTypes } from 'react';
import TextField from 'material-ui/TextField';
import CurrentContext from 'utils/currentContext'
import ImageSelector from 'utils/imageSelector'
import RaisedButton from 'material-ui/RaisedButton';
import Checkbox from 'material-ui/Checkbox';
import { Link } from 'react-router';

import styles from '../Shared/LoginForm.scss';
import { buttonStyle, fieldStyle, checkboxStyle } from '../Shared/FormStyle';

const PreviewIcon = ImageSelector(CurrentContext.theme, 'svgs/preview.svg');
const warning = ImageSelector(CurrentContext.theme, 'pending.png');

const {string, func} = PropTypes;

class SignInForm extends Component {
  render() {
    return (
      <div>
        <div className={styles.wrong_password}>
          <div className={styles.warning_img}>
            <img src={warning} alt="warning" />
          </div>
          <div className={styles.wrong_pass_text}>
            <div>
              <div>Wrong Password!</div>
              <div>Password does not match provided email.</div>
            </div>
          </div>
        </div>
        <div className={styles.form_containter}>
          <div className={styles.title} style={{ paddingTop: 70 }}>
            <div data-spec='title' >{this.props.title}</div>
            <div data-spec='subtitle'>{this.props.subtitle}</div>
          </div>
          <div className={styles.field_containter}>
            <div className={styles.field_title}>E MAIL</div>
            <TextField
              inputStyle={fieldStyle.inputStyle}
              underlineFocusStyle={fieldStyle.underlineFocusStyle}
              underlineStyle={fieldStyle.underlineStyle}
              fullWidth={true}
              style={fieldStyle.style}
              id="email" />
            <br />
            <div className={styles.field_title}>PASSWORD</div>
            <TextField
              inputStyle={fieldStyle.inputStyle}
              underlineFocusStyle={fieldStyle.underlineFocusStyle}
              underlineStyle={fieldStyle.underlineStyle}
              fullWidth={true}
              style={fieldStyle.style}
              type="password"
              id="password" /><br />
            <div className={styles.checkbox_wrapper}>
              <Checkbox
                data-spec='remember-me'
                label="Remember me"
                className={styles.checkbox}
                iconStyle={checkboxStyle.iconStyle}
                />
              <div className={styles.forgot_password}><Link data-spec='forgot-password'>Forgot password?</Link></div>
            </div>
          </div>
          <div className={styles.button_containter}>
            <RaisedButton data-spec='login-button' overlayStyle={buttonStyle.primary.overlayStyle} buttonStyle={buttonStyle.primary.style} key="Login" label="Log In" primary={true} fullWidth={true} onClick={this.props.login} />
            <RaisedButton data-spec='request-access-button' buttonStyle={buttonStyle.secondary.style} style={buttonStyle.secondary.style} labelStyle={buttonStyle.secondary.labelStyle} key="Request" label="Request Access" fullWidth={true} onClick={this.props.request} />
          </div>
        </div>
      </div>
    );
  }
}

SignInForm.defaultProps = {
  title: 'Log In',
  subtitle: 'Please enter your email and password.'
}

SignInForm.propTypes = {
  title: string,
  subtitle: string,
  login: func,
  request: func
}

export default SignInForm;