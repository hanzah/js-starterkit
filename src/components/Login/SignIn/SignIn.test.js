import React from 'react';
import { mount, shallow } from 'enzyme';

import SignIn from './SignIn';

describe('<SignIn/>', () => {

  it('should render a component', () => {
    const wrapper = shallow(<SignIn />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have Login component', () => {
    const wrapper = shallow(<SignIn />);
    const Login = wrapper.find('Login');
    expect(Login).toBeDefined();
  });

});