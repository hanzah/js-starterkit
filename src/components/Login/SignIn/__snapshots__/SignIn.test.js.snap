exports[`<SignIn/> should render a component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    data-spec="sign-in-screen">
    <Login
        form={
            <SignInForm
                subtitle="Please enter your email and password."
                title="Log In" />
        }
        logo={
            <LoginLogo
                productName="PRODUCT TITLE" />
        } />
</div>,
  "nodes": Array [
    <div
      data-spec="sign-in-screen">
      <Login
            form={
                  <SignInForm
                        subtitle="Please enter your email and password."
                        title="Log In" />
            }
            logo={
                  <LoginLogo
                        productName="PRODUCT TITLE" />
            } />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <SignIn />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          data-spec="sign-in-screen">
          <Login
                    form={
                              <SignInForm
                                        subtitle="Please enter your email and password."
                                        title="Log In" />
                    }
                    logo={
                              <LoginLogo
                                        productName="PRODUCT TITLE" />
                    } />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          data-spec="sign-in-screen">
          <Login
                    form={
                              <SignInForm
                                        subtitle="Please enter your email and password."
                                        title="Log In" />
                    }
                    logo={
                              <LoginLogo
                                        productName="PRODUCT TITLE" />
                    } />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SignIn />,
}
`;
